csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -144,20 +144,40 @@ public static string NestPaths(ILogger logger, string? outerpath, string innerpa
             return nested;
         }
 
+        private static string? tempFolderPath = null;
+        private static readonly object lockObject = new();
+
         public static string GetTemporaryWorkingDirectory(Func<string, string?> getEnvironmentVariable, string lang, out bool shouldCleanUp)
         {
             shouldCleanUp = false;
             var tempFolder = getEnvironmentVariable($"CODEQL_EXTRACTOR_{lang}_SCRATCH_DIR");
+            if (!string.IsNullOrEmpty(tempFolder))
+            {
+                return tempFolder;
+            }
 
-            if (string.IsNullOrEmpty(tempFolder))
+            if (!string.IsNullOrEmpty(tempFolderPath))
```

Comment:
I don't think [double checked locking](https://en.wikipedia.org/wiki/Double-checked_locking) works in C#, at least the field will then have to be `volatile`. In all cases, I think it's better to use `Lazy<string>`.